pkg/common/hubconfig/hubconfig_test.go

```diff
@@ -19,63 +19,95 @@ func TestPrepareHubConfig(t *testing.T) {
 			tlsClientInsecure:    false,
-			wantErr:              true,
+			validate: func(t *testing.T, config *rest.Config, err error) {
+				if err == nil {
+					t.Errorf("expect return error if CONFIG_PATH not present")
+				}
+			},
 		},
 		{
 			name:                 "environment variable `HUB_CERTIFICATE_AUTHORITY` is not present when tlsClientInsecure is false",
 			environmentVariables: map[string]string{hubServerURLEnvKey: fakeHubhubServerURLEnvVal, tokenConfigPathEnvKey: fakeConfigtokenConfigPathEnvVal},
 			tlsClientInsecure:    false,
-			wantErr:              true,
+			validate: func(t *testing.T, config *rest.Config, err error) {
+				if err != nil {
+					t.Errorf("not expect error but actually get error %s", err)
+				}
+				cmp.Equal(config, &rest.Config{
+					BearerTokenFile: fakeConfigtokenConfigPathEnvVal,
+					Host:            fakeHubhubServerURLEnvVal,
+					TLSClientConfig: rest.TLSClientConfig{
+						Insecure: true,
+					},
+				})
+			},
 		},
 		{
 			name:                 "environment variable `HUB_CERTIFICATE_AUTHORITY` is not present when tlsClientInsecure is true",
 			environmentVariables: map[string]string{hubServerURLEnvKey: fakeHubhubServerURLEnvVal, tokenConfigPathEnvKey: fakeConfigtokenConfigPathEnvVal},
 			tlsClientInsecure:    true,
-			wantErr:              false,
+			validate: func(t *testing.T, config *rest.Config, err error) {
+				if err != nil {
+					t.Errorf("not expect error but actually get error %s", err)
+				}
+				cmp.Equal(config, &rest.Config{
+					BearerTokenFile: fakeConfigtokenConfigPathEnvVal,
+					Host:            fakeHubhubServerURLEnvVal,
+					TLSClientConfig: rest.TLSClientConfig{
+						Insecure: true,
+					},
+				})
+			},
 		},
 		{
 			name:                 "hub configuration preparation is done when all requirements meet",
 			environmentVariables: map[string]string{hubServerURLEnvKey: fakeHubhubServerURLEnvVal, tokenConfigPathEnvKey: fakeConfigtokenConfigPathEnvVal, hubCAEnvKey: fakeCerhubCAEnvVal},
 			tlsClientInsecure:    false,
-			wantErr:              false,
+			validate: func(t *testing.T, config *rest.Config, err error) {
+				if err != nil {
+					t.Errorf("not expect error but actually get error %s", err)
+				}
+				cmp.Equal(config, &rest.Config{
+					BearerTokenFile: fakeConfigtokenConfigPathEnvVal,
+					Host:            fakeHubhubServerURLEnvVal,
+					TLSClientConfig: rest.TLSClientConfig{
+						Insecure: true,
+						CAData:   fakeCerhubCA,
+					},
+				})
+			},
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			// remove all environment variables related to this test
-			cleanupFunc()
-
 			for envKey, envVal := range tc.environmentVariables {
-				os.Setenv(envKey, envVal)
+				t.Setenv(envKey, envVal)
 				if envKey == tokenConfigPathEnvKey {
 					if _, err := os.Create(fakeConfigtokenConfigPathEnvVal); err != nil {
```

Comment:
Shouldn't call `os.Create` in unit tests - can this now be removed, especially if using testdata/ ?